c10/test/util/intrusive_ptr_test.cpp

```diff
@@ -11,6 +11,8 @@ using c10::intrusive_ptr_target;
 using c10::make_intrusive;
 using c10::weak_intrusive_ptr;
 
+#pragma GCC diagnostic ignored "-Wpragmas"
+#pragma GCC diagnostic ignored "-Wunknown-warning-option"
```

Comment:
FWIW, I've kind of decided I don't like these diagnostic pragmas. They're fiddly and hard to get right.

Reply:
Yeah, I have mixed feelings about them. They're necessary to appease `-Werror` together with a warning option that generally makes sense to be enabled, but your concrete code has to violate it and can't work around it. On one side, `-Werror` and the warning option make sense for most cases, and have the chance of catching bugs elsewhere, but not for this one instance, so you add the pragma...but that has its own issues as you mentioned.